# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: uriloader/exthandler/win/nsMIMEInfoWin.cpp
# Commit: b9d9cb83485d
# Full Hash: b9d9cb83485d6e256b94d1ff0fac2841f2d84e9e
# Author: Jared Wein <jwein@mozilla.com>
# Date: 2020-06-06 09:46:03
# Description:
#   Bug 1643673 - Guard against a null default application. r=tkikuchi
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78551
# ==============================================================================

diff -r c272f2f9aaac -r b9d9cb83485d uriloader/exthandler/win/nsMIMEInfoWin.cpp
--- a/uriloader/exthandler/win/nsMIMEInfoWin.cpp	Fri Jun 05 19:08:49 2020 +0000
+++ b/uriloader/exthandler/win/nsMIMEInfoWin.cpp	Fri Jun 05 18:19:12 2020 +0000
@@ -94,7 +94,8 @@
                "nsMIMEInfoBase should have mClass == eMIMEInfo");
 
   if (mPreferredAction == useSystemDefault) {
-    if (StaticPrefs::browser_pdf_launchDefaultEdgeAsApp()) {
+    if (mDefaultApplication &&
+        StaticPrefs::browser_pdf_launchDefaultEdgeAsApp()) {
       // Since Edgium is the default PDF handler, if we're using the OS default
       // and it's Edgium prefer it's app mode so it operates as a PDF viewer
       // (without browser toolbars). Bug 1632277.
